Carpet/src/Comm.cc

```diff
@@ -242,12 +242,10 @@ static void ProlongateGroupBoundaries(const cGH *cctkGH,
     for (astate state = static_cast<astate>(0); state != state_done;
          state = static_cast<astate>(static_cast<int>(state) + 1)) {
       ostringstream name1;
-      name1 << "comm_state[" << timers.size() << "]"
-            << "." << tostring(state) << ".user";
+      name1 << "comm_state[" << timers.size() << "].user." << tostring(state);
       timers.push_back(new Timers::Timer(name1.str()));
       ostringstream name2;
-      name2 << "comm_state[" << timers.size() << "]"
-            << "." << tostring(state) << ".step";
+      name2 << "comm_state[" << timers.size() << "].step." << tostring(state);
```

Comment:
This seems to change the name of timers from "state.user" and "state.step" to "user.state" and "user.step" which is a non-backwards compatible, user visible change for anyone who wants to parse the output files for a particular user. Since the change is only cosmetic, I would rather keep the old names. This affects only the first of the two commits in the branch of the pull request.

Reply:
@eschnett says

This change tries to ensure that the important part of the timer name remains readable when the timer name is truncated, which happens in thorn TimerReport. Moving "user" and "step" (see below) to an earlier position in the string helps a lot.

Reply:
Wouldn't it then be better to remove the (arbitrary) restriction on timer name length from TimerReport? Eg use something better than (more or less) char [n_timers][100] timerbuffer?

I am aware this is very minor and will only affect code that parses for these timer names which is intrinsically "termporary" since it is performance analysis scripts and the like.

Reply:
See https://bitbucket.org/einsteintoolkit/tickets/issues/2268/timerreport-allow-arbirary-long-timer for a patch that removes the limit from TimerReport.

I’ll create version of this pull request that:

* removes the omp parallel more cleanly

* does not change the timer names

once that ticket has been reviewed and the pull request acted on.